app/src/main/java/com/example/moviewithfragments/request/Service.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.moviewithfragments.request
+
+import com.example.moviewithfragments.utils.Credentials
+import com.example.moviewithfragments.utils.MovieApi
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class Service {
+    var credentials = Credentials()
```

Comment:
when ever you call `Service` it will create new Retrofit client. Instead use singleton `object`
we have seen this in the example I updated while explaining